### PR TITLE
Add quickstart section to root-level README

### LINKED ISSUE #249: Consolidate quickstart guides
It looks like there are a variety of quickstart guides (e.g. [this one](https://github.com/operator-framework/rukpak/blob/main/docs/plain-bundle-spec.md#quickstart) and [this one](https://github.com/operator-framework/rukpak/tree/main/internal/provisioner/plain)) and it would be a bit clearer to consolidate them to streamline the first time UX.

# Acceptance Criteria
- [ ] Clear step-by-step guide for getting started with RukPak, ideally in the main README

### PR DESCRIPTION
**Pull request by exdx**
Closes #249

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -21,6 +21,23 @@ the content.
 The RukPak project is community driven and is part of the broader Kubernetes ecosystem. New contributors are welcome and
 highly encouraged. See the [contributing guidelines](CONTRIBUTING.md) to get started.
 
+## Quickstart
+
+The RukPak project consists of a series of controllers, known as provisioners, that install and manage content on a
```
Comment (timflannagan) [inline README.md:26]: Link to the provisioner header in the README here?

File: README.md [hunk 2]
```diff
@@ -21,6 +21,23 @@ the content.
...
+this provisioner, see [the quickstart section](./internal/provisioner/plain/README.md#Running-locally) of the plain
+provisioner README. There will be other provisioners added to the RukPak project that support different content types.
+
+The plain provisioner is able to source and unpack plain bundles. To learn more about the plain bundle format,
+see [the plain bundle spec](./docs/plain-bundle-spec.md).
+
+This project uses GitHub issues and milestones to prioritize and keep track of ongoing work. To see the current state of
+the project, checkout the [open issues](https://github.com/operator-framework/rukpak/issues) and
+[recent milestones](https://github.com/operator-framework/rukpak/milestones). There is a lot of development work ongoing
+and lots of opportunities to get involved and contribute!
```
Comment (timflannagan) [inline README.md:39]: Link to the CONTRIBUTING.md documentation?

**Review by timflannagan — APPROVED** [review]

**Comment by timflannagan** [comment]
Nice work as always - only question: we're referencing an issue that's centered around consolidating quickstart guides yet it seems like these changes are more geared towards an introduction of the plain provisioner, vs. actually consolidating quickstart guides. Should we create a different ticket here instead?

File: README.md [hunk 3]
```diff
@@ -21,6 +21,23 @@ the content.
 The RukPak project is community driven and is part of the broader Kubernetes ecosystem. New contributors are welcome and
 highly encouraged. See the [contributing guidelines](CONTRIBUTING.md) to get started.
 
+## Quickstart
+
+The RukPak project consists of a series of controllers, known as provisioners, that install and manage content on a
+Kubernetes cluster. See below for a more detailed look into the APIs that RukPak provides.
```
Comment (tylerslaton) [inline README.md:27]: Might be worth linking to where this starts in the document so that a user can easily navigate to it.

File: README.md [hunk 4]
```diff
@@ -21,6 +21,23 @@ the content.
...
+
+The provisioner currently implemented and bundled with RukPak is known as the plain provisioner. To get started with
+this provisioner, see [the quickstart section](./internal/provisioner/plain/README.md#Running-locally) of the plain
```
Comment (oceanc80) [inline README.md:30]: Might be worth specifying that this quickstart is for running locally vs. on a cluster.

**Comment by exdx** [comment]
> Nice work as always - only question: we're referencing an issue that's centered around consolidating quickstart guides yet it seems like these changes are more geared towards an introduction of the plain provisioner, vs. actually consolidating quickstart guides. Should we create a different ticket here instead?

My impression of the issue was that the quickstarts were scattered and not easy to navigate, and that we should introduce a section to the root-level README that helps to navigate the different docs. I didn't think that the intent was to consolidate necessarily, since each doc is valuable on its own and has its own concept of quickstart/demo. The goal of the new section being introduced here is to make it easier for a first time user to onboard and navigate the different documents. Unless I misunderstood the spirit of the original issue, I don't think a new ticket is necessary -- maybe @oceanc80 could clarify if this PR doesn't meet the requirements?

**Comment by oceanc80** [comment]
@exdx I think once we have more provisioners it'd be nice to surface examples of running locally and on a cluster with the plain provisioner directly in this readme, but I think for now as long as we have the link to the existing docs, that should be fine. We could maybe also add the [cluster quickstart link](https://github.com/operator-framework/rukpak/blob/main/docs/plain-bundle-spec.md#quickstart) as well just so both the local and cluster quickstarts are linked from this readme, but overall I think this is good for now.

**Review by oceanc80 — APPROVED** [review]

**Comment by exdx** [comment]
> @exdx I think once we have more provisioners it'd be nice to surface examples of running locally and on a cluster with the plain provisioner directly in this readme, but I think for now as long as we have the link to the existing docs, that should be fine. We could maybe also add the [cluster quickstart link](https://github.com/operator-framework/rukpak/blob/main/docs/plain-bundle-spec.md#quickstart) as well just so both the local and cluster quickstarts are linked from this readme, but overall I think this is good for now.

I agree, yes once we have more provisioners we will have to reorganize the docs quite a bit, right now they are overly specified towards the plain provisioner and plain bundle format. 

For the plain bundle spec, I think it's still targeting a local installation, it just describes how to build a plain bundle image, which is slightly tangential to running rukpak locally or on a real cluster. I think you bring up a good point, that we should probably have some kind of target and documentation around installing rukpak on a real (non-kind) cluster. Right now we only have a `run` target to deploy rukpak which assumes its dealing with a kind cluster (we removed other `deploy` and `install` targets because they were causing unnecessary confusion). I will go ahead and open an issue for that.

**Comment by exdx** [comment]
#312

**Pull request merged**